server/render.js

```diff
@@ -147,17 +148,27 @@ export async function serveStaticWithGzip (req, res, path) {
     return serveStatic(req, res, path)
   }
 
+  const gzipPath = `${path}.gz`
+
   try {
-    const gzipPath = `${path}.gz`
-    res.setHeader('Content-Encoding', 'gzip')
-    await serveStatic(req, res, gzipPath)
+    // We need to check the existance of the gzipPath.
+    // Getting `ENOENT` error from the `serveStatic` is inconsistance and
+    // didn't work on all the cases.
+    //
+    // And this won't give us a race condition because we know that
+    // we don't add gzipped files at runtime.
+    await fs.stat(gzipPath)
   } catch (ex) {
     if (ex.code === 'ENOENT') {
-      res.removeHeader('Content-Encoding')
+      // Seems like there's no gzipped file. Let's serve the uncompressed file.
```

Comment:
Specially in dev mode.
We can remove this check when we get https://github.com/zeit/next.js/pull/657